### ISSUE TITLE
Detect xhyve/bhyve guest

### ISSUE THREAD

**Issue by Hipska** [jrgp/linfo#47]
Linfo detects guest of xhyve and bhyve as being QEMU/KVM.

This seems to be a generic problem as I have found tickets for Salt and Puppet/facter that state the same also.

**Comment by jrgp** [comment]
This makes sense as these two new hypervisors use libvirt like qemu does. It'd be easiest of they loaded a custom kernel module that I could look for but I don't think they do.

I'm open to suggestions. Maybe there's a secret file in /sys or /proc that could be informative here.

**Comment by Hipska** [comment]
Does this makes some sense?
```
root@ee2f73bc1f20:/# grep -ri hyve /sys/devices/virtual/dmi/id
/sys/devices/virtual/dmi/id/product_name:BHYVE
/sys/devices/virtual/dmi/id/modalias:dmi:bvnBHYVE:bvr1.00:bd03/14/2014:svn:pnBHYVE:pvr1.0:cvn:ct2:cvr1.0:
/sys/devices/virtual/dmi/id/bios_vendor:BHYVE
/sys/devices/virtual/dmi/id/uevent:MODALIAS=dmi:bvnBHYVE:bvr1.00:bd03/14/2014:svn:pnBHYVE:pvr1.0:cvn:ct2:cvr1.0:
```

**Comment by jrgp** [comment]
Yes, checking if that one of those files exists and has the name of the hypervisor in it would definitely make sense.

**Comment by Hipska** [comment]
Yeah, but it doesn't seem good enough as this is taken from an xhyve Guest. (actually it is on a container from Docker for mac)

**Comment by jrgp** [comment]
Yeah I recall a PR on the xhyve project to make it identify itself as xhyve not bhyve

**Comment by jrgp** [comment]
Maybe see if xhyve is written anywhere

**Comment by Hipska** [comment]
At least not in `/sys`, but you could identify them as bhyve/xhyve guest and use this logo for example: ![bhyve logo](http://www.bhyve.org/static/bhyve.png)

**Comment by jrgp** [comment]
That makes sense. Would you be able to make a PR for this?

**Comment by jrgp** [comment]
^

**Comment by jrgp** [comment]
^

**Comment by Hipska** [comment]
Hi sorry for late comment, but I think you could guess that yourself already that I currently have no time to spend on this. Could you please do it yourself? It doesn't seem a lot of work.
